### DISCUSSION TITLE
[QUESTION] Publish MQTT into BSB_LAN_custom_setup.h file

### DISCUSSION THREAD

**Discussion by Turbox35** [fredlcore/BSB-LAN#604] [General]
Hello,

This is not a bug or request, the problem is on my side.
I would like to publish MQTT message at the end of the setup (run one time at the end of the boot) and I use the command:
`MQTTPubSubClient->publish(Topic.c_str(), output, n, true);`
But it fails. 
`BSB-LAN-master\BSB_LAN\BSB_LAN_custom_setup.h:50:66: error: no matching function for call to 'PubSubClient::publish(const char*, String&, size_t&, bool)'`
Is it due to the MQTT client which is not already declared? Or I miss something?

Many thanks.

**Comment by fredlcore** [comment]
Well, the error message is clear: You are calling a function that expects other parameters than you are sending. If you look at PubSubClient.h, you'll see that there are four different ways to call "publish":
```
   boolean publish(const char* topic, const char* payload);
   boolean publish(const char* topic, const char* payload, boolean retained);
   boolean publish(const char* topic, const uint8_t * payload, unsigned int plength);
   boolean publish(const char* topic, const uint8_t * payload, unsigned int plength, boolean retained);
```
Yours is of the structure `const char*, String&, size_t&, bool`. So it doesn't expect your string but rather a uint8 pointer.

**Comment by Turbox35** [comment]
Thank you very much, it makes sense now.
Changing to `MQTTPubSubClient->publish(Topic.c_str(), (uint8_t*)output.c_str(), n, true);` make no compilation error but the ESP-32 enter in a boot loop when it reaches this line:

```
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4017c8ff  PS      : 0x00060d30  A0      : 0x800e3fb6  A1      : 0x3ffb1f00  
A2      : 0x00000000  A3      : 0x3ffc484c  A4      : 0x000000ff  A5      : 0x0000ff00  
A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x800ecffc  A9      : 0x3ffb1ee0  
A10     : 0x00000002  A11     : 0x3ffc484c  A12     : 0x00000000  A13     : 0x00000001  
A14     : 0x00ff0000  A15     : 0xff000000  SAR     : 0x00000018  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000008  LBEG    : 0x4008b22d  LEND    : 0x4008b23d  LCOUNT  : 0xffffffff  


Backtrace: 0x4017c8fc:0x3ffb1f00 0x400e3fb3:0x3ffb1f20 0x400e1fd9:0x3ffb1f40 0x400efe0e:0x3ffb2290




ELF file SHA256: 598961948f6eba6e

Rebooting...
```

**Comment by fredlcore** [comment]
I don't know if you can simply convert a string to a uint8_t like that, so maybe that's the problem. But since nothing of your problems has anything to do with BSB-LAN, I'll close this here. You can certainly find help in more relevant coding forums.

**Comment by Turbox35** [comment]
I understand, you can close my question.
Have a nice day.
